src/underactuated/pyplot_visualizer.py

```diff
@@ -28,23 +28,33 @@ class PyPlotVisualizer(LeafSystem):
     '''
 
     def __init__(self, draw_timestep=0.033333, facecolor=[1, 1, 1],
-                 figsize=None):
+                 figsize=None, fig=None, ax=None):
         LeafSystem.__init__(self)
 
         self.set_name('pyplot_visualization')
         self.timestep = draw_timestep
         self._DeclarePeriodicPublish(draw_timestep, 0.0)
 
-        (self.fig, self.ax) = plt.subplots(facecolor=facecolor,
-                                           figsize=figsize)
+        if ax is None:
+            if fig is not None:
+                print "Error: supplied fig but not ax."
+            (self.fig, self.ax) = plt.subplots(facecolor=facecolor,
+                                               figsize=figsize)
+        else:
+            self.fig = fig
+            self.ax = ax
+
         self.ax.axis('equal')
         self.ax.axis('off')
-        self.fig.show()
+
+        if self.fig is not None:
+            self.fig.show()
```

Comment:
why let it get this far without creating the figure/axis? 

furthemore, if you require that one sends both the fig and the axis, then I think that's redundant?  can't you just send the axis, and figure out the figure from the axis?  

the only case that requires this conditional right now is if someone passes in ax not none an fig=none, which seems non-sensible.